chisel/src/dispatcher.rs

```diff
@@ -379,6 +379,32 @@ impl ChiselDispatcher {
                     DispatchResult::CommandFailed(Self::make_error("Session not present."))
                 }
             }
+            ChiselCommand::Calldata => {
+                if let Some(session_source) = self.session.session_source.as_mut() {
+                    if args.is_empty() {
+                        session_source.config.calldata = None;
+                        return DispatchResult::CommandSuccess(Some("Calldata cleared.".to_string()))
+                    }
+
+                    let calldata =
+                        hex::decode(args[0].trim_matches(|c: char| c.is_whitespace() || c == '"'));
```

Comment:
decode does not strip the 0x prefix, which could be present since user input?

so we need a trim_start_matches here for 0x as well?

Reply:
good catch, will adjust